### PR TITLE
#4261 - fixed text overflowing on second line on small devices

### PR DESCRIPTION
**Pull request by florianf22**
**Related issue(s):**
* Fixes https://github.com/scandipwa/scandipwa/issues/4261

**Problem:**
* The problem was that when going on small devices, terms and conditions text were stretching on two lines and so did the '-' symbol.

**In this PR:**
* used screen variable to render corresponding layout. this PR had already been opened here - https://github.com/scandipwa/scandipwa/pull/4523 but because I deleted branch and re-opened, it was deleted as well. Re-created now

### CODE REVIEW

**Comment by aleksandrakorolova** [comment]
Please, merge master in this PR

**Comment by florianf22** [comment]
What do you mean merge master?

**Comment by aleksandrakorolova** [comment]
I mean - Merge branch 'master' into issue as you did [here](https://github.com/scandipwa/scandipwa/pull/4447)

**Comment by aleksandrakorolova** [comment]
Please, fix amplify
Screenshot 2022-03-29 at 13 32 40

**Comment by florianf22** [comment]
link fixed

**Comment by aleksandrakorolova** [comment]
No, Link isn't fixed 
Screenshot 2022-03-30 at 08 08 21

**Comment by aleksandrakorolova** [comment]
```
                                 # Starting phase: preBuild
                                 # Executing command: nvm install 14
2022-03-29T10:56:26.321Z [INFO]: Downloading and installing node v14.19.1...
2022-03-29T10:56:26.404Z [WARNING]: Downloading https://nodejs.org/dist/v14.19.1/node-v14.19.1-linux-x64.tar.gz...
2022-03-29T10:56:26.522Z [WARNING]: ##################################                                        47.8%
2022-03-29T10:56:26.578Z [WARNING]: ################################################
2022-03-29T10:56:26.578Z [WARNING]: ######################## 100.0%
2022-03-29T10:56:26.594Z [WARNING]: Computing checksum with sha256sum
2022-03-29T10:56:26.682Z [WARNING]: Checksums matched!
2022-03-29T10:56:27.716Z [INFO]: Now using node v14.19.1 (npm v6.14.16)
2022-03-29T10:56:27.746Z [INFO]: Installing default global packages from /root/.nvm/default-packages...
2022-03-29T10:56:27.746Z [INFO]: npm install -g --quiet yarn@1.22.0
2022-03-29T10:56:29.082Z [INFO]: /root/.nvm/versions/node/v14.19.1/bin/yarn -> /root/.nvm/versions/node/v14.19.1/lib/node_modules/yarn/bin/yarn.js
2022-03-29T10:56:29.084Z [INFO]: /root/.nvm/versions/node/v14.19.1/bin/yarnpkg -> /root/.nvm/versions/node/v14.19.1/lib/node_modules/yarn/bin/yarn.js
2022-03-29T10:56:29.085Z [INFO]: + yarn@1.22.0
                                 added 1 package in 1.014s
2022-03-29T10:56:29.128Z [INFO]: # Executing command: nvm use 14
2022-03-29T10:56:29.366Z [INFO]: Now using node v14.19.1 (npm v6.14.16)
2022-03-29T10:56:29.367Z [INFO]: # Executing command: yarn install
2022-03-29T10:56:29.763Z [INFO]: yarn install v1.18.0
2022-03-29T10:56:29.896Z [INFO]: [1/4] Resolving packages...
2022-03-29T10:56:30.750Z [INFO]: [2/4] Fetching packages...
2022-03-29T10:56:55.747Z [WARNING]: warning scandipwa-development-toolkit-vscode@1.0.25: The engine "vscode" appears to be invalid.
2022-03-29T10:56:55.749Z [INFO]: info fsevents@2.1.3: The platform "linux" is incompatible with this module.
2022-03-29T10:56:55.750Z [INFO]: info "fsevents@2.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.
2022-03-29T10:56:55.750Z [INFO]: info fsevents@2.3.2: The platform "linux" is incompatible with this module.
                                 info "fsevents@2.3.2" is an optional dependency and failed compatibility check. Excluding it from installation.
2022-03-29T10:56:55.754Z [INFO]: info fsevents@1.2.13: The platform "linux" is incompatible with this module.
2022-03-29T10:56:55.755Z [INFO]: info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
2022-03-29T10:56:55.761Z [INFO]: [3/4] Linking dependencies...
2022-03-29T10:56:55.765Z [WARNING]: warning "lerna > @lerna/version > @lerna/github-client > @octokit/rest > @octokit/plugin-request-log@1.0.4" has unmet peer dependency "@octokit/core@>=3".
2022-03-29T10:56:55.766Z [WARNING]: warning " > @scandipwa/router@0.0.8" has incorrect peer dependency "@scandipwa/framework@^0.0.1".
2022-03-29T10:56:55.766Z [WARNING]: warning " > @scandipwa/scandipwa-scripts@2.4.19" has unmet peer dependency "@tilework/mosaic@0.1.4".
                                    warning " > @scandipwa/service-worker@0.0.17" has unmet peer dependency "@tilework/mosaic-craco@0.0.5".
2022-03-29T10:56:55.766Z [WARNING]: warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/router > react-router@5.2.1" has unmet peer dependency "react@>=15".
                                    warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/router > react-router-dom@5.3.0" has unmet peer dependency "react@>=15".
2022-03-29T10:56:55.767Z [WARNING]: warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/stylelint-config > stylelint-use-logical@1.1.0" has incorrect peer dependency "stylelint@^9.6.0".
2022-03-29T10:56:55.768Z [WARNING]: warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/stylelint-config > stylelint-webpack-plugin@3.1.0" has unmet peer dependency "webpack@^5.0.0".
2022-03-29T10:56:55.768Z [WARNING]: warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/scandipwa-development-toolkit-core > @typescript-eslint/eslint-plugin@2.34.0" has incorrect peer dependency "eslint@^5.0.0 || ^6.0.0".
2022-03-29T10:56:55.768Z [WARNING]: warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/scandipwa-development-toolkit-core > @typescript-eslint/parser@2.34.0" has incorrect peer dependency "eslint@^5.0.0 || ^6.0.0".
2022-03-29T10:56:55.768Z [WARNING]: warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/router > react-router > mini-create-react-context@0.4.1" has unmet peer dependency "react@^0.14.0 || ^15.0.0 || ^16.0.0 || ^17.0.0".
2022-03-29T10:56:55.769Z [WARNING]: warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/scandipwa-scripts > @tilework/mosaic-cra-scripts > @tilework/mosaic-config-injectors@0.1.17" has unmet peer dependency "@babel/core@^7.14.0".
2022-03-29T10:56:55.769Z [WARNING]: warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/scandipwa-scripts > @tilework/mosaic-cra-scripts > @tilework/mosaic-craco@0.0.5" has incorrect peer dependency "react-scripts@4.0.1".
                                    warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/scandipwa-scripts > @tilework/mosaic-cra-scripts > babel-jest@26.6.3" has unmet peer dependency "@babel/core@^7.0.0".
                                    warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/scandipwa-scripts > @tilework/mosaic-cra-scripts > craco-sass-resources-loader@1.1.0" has unmet peer dependency "@craco/craco@^3.3.1".
2022-03-29T10:56:55.770Z [WARNING]: warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/scandipwa-scripts > @tilework/mosaic-cra-scripts > craco-sass-resources-loader@1.1.0" has incorrect peer dependency "react-scripts@^2.1.3".
                                    warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/scandipwa-scripts > @tilework/mosaic-cra-scripts > progress-bar-webpack-plugin@2.1.0" has unmet peer dependency "webpack@^1.3.0 || ^2 || ^3 || ^4 || ^5".
                                    warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/scandipwa-scripts > @tilework/mosaic-cra-scripts > react-scripts@4.0.3" has unmet peer dependency "react@>= 16".
2022-03-29T10:56:55.770Z [WARNING]: warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/scandipwa-scripts > @tilework/mosaic-cra-scripts > @tilework/mosaic-config-injectors > @babel/plugin-transform-arrow-functions@7.13.0" has unmet peer dependency "@babel/core@^7.0.0-0".
                                    warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/scandipwa-scripts > @tilework/mosaic-cra-scripts > @tilework/mosaic-config-injectors > @babel/plugin-transform-async-to-generator@7.13.0" has unmet peer dependency "@babel/core@^7.0.0-0".
2022-03-29T10:56:55.770Z [WARNING]: warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/scandipwa-scripts > @tilework/mosaic-cra-scripts > @tilework/mosaic-config-injectors > babel-loader@8.1.0" has unmet peer dependency "@babel/core@^7.0.0".
                                    warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/scandipwa-scripts > @tilework/mosaic-cra-scripts > @tilework/mosaic-config-injectors > babel-loader@8.1.0" has unmet peer dependency "webpack@>=2".
                                    warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/scandipwa-scripts > @tilework/mosaic-cra-scripts > @tilework/mosaic-test-utils > @testing-library/react@11.2.7" has unmet peer dependency "react-dom@*".
                                    warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/scandipwa-scripts > @tilework/mosaic-cra-scripts > babel-jest > babel-preset-jest@26.6.2" has unmet peer dependency "@babel/core@^7.0.0".
2022-03-29T10:56:55.772Z [WARNING]: warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/scandipwa-scripts > @tilework/mosaic-cra-scripts > @tilework/mosaic-test-utils > @tilework/mosaic-nextjs-scripts > next@10.0.5" has unmet peer dependency "react-dom@^16.6.0 || ^17".
2022-03-29T10:56:55.773Z [WARNING]: warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/scandipwa-scripts > @tilework/mosaic-cra-scripts > babel-jest > babel-preset-jest > babel-preset-current-node-syntax@1.0.1" has unmet peer dependency "@babel/core@^7.0.0".
2022-03-29T10:56:55.774Z [WARNING]: warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/scandipwa-scripts > @tilework/mosaic-cra-scripts > @tilework/mosaic-test-utils > @tilework/mosaic-nextjs-scripts > next > @next/react-dev-overlay@10.0.5" has unmet peer dependency "react-dom@^16.9.0 || ^17".
2022-03-29T10:56:55.777Z [WARNING]: warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/scandipwa-scripts > @tilework/mosaic-cra-scripts > babel-jest > babel-preset-jest > babel-preset-current-node-syntax > @babel/plugin-syntax-bigint@7.8.3" has unmet peer dependency "@babel/core@^7.0.0-0".
2022-03-29T10:56:55.777Z [WARNING]: warning "workspace-aggregator-db2d53f9-fe8a-4079-a2a9-48053ed0fe57 > @scandipwa/scandipwa-scripts > @tilework/mosaic-cra-scripts > babel-jest > babel-preset-jest > babel-preset-current-node-syntax > @babel/plugin-syntax-import-meta@7.10.4" has unmet peer dependency "@babel/core@^7.0.0-0".
2022-03-29T10:56:55.834Z [WARNING]: warning Workspaces can only be enabled in private projects.
2022-03-29T10:57:25.962Z [INFO]: [4/4] Building fresh packages...
2022-03-29T10:57:28.838Z [INFO]: Done in 59.08s.
2022-03-29T10:57:28.872Z [INFO]: # Completed phase: preBuild
                                 # Starting phase: build
2022-03-29T10:57:28.873Z [INFO]: # Executing command: yarn run build
2022-03-29T10:57:29.174Z [INFO]: yarn run v1.18.0
2022-03-29T10:57:29.210Z [INFO]: $ scandipwa-scripts build
2022-03-29T10:57:33.577Z [WARNING]: Browserslist: caniuse-lite is outdated. Please run:
                                    npx browserslist@latest --update-db
                                    Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
2022-03-29T10:57:33.693Z [INFO]: Creating an optimized production build...
2022-03-29T10:59:41.666Z [INFO]: [31mFailed to compile.[39m
                                 [31m[39m
2022-03-29T10:59:41.668Z [INFO]: [4msrc/component/CheckoutBilling/CheckoutBilling.style.scss[24m
                                 [2m12:2[22m  [31m[31m✖[39m  Expected indentation of 0 spaces  [2mindentation[22m
2022-03-29T10:59:41.751Z [INFO]: Done in 132.58s.
2022-03-29T10:59:41.758Z [INFO]: # Completed phase: build
2022-03-29T10:59:41.759Z [INFO]: ## Build completed successfully
2022-03-29T10:59:41.760Z [INFO]: # Starting caching...
2022-03-29T10:59:41.844Z [INFO]: # Creating cache artifact...
2022-03-29T10:59:55.715Z [INFO]: # Cache artifact is: 650MB
2022-03-29T10:59:55.797Z [INFO]: # Uploading cache artifact...
2022-03-29T11:00:03.300Z [INFO]: # Caching completed
2022-03-29T11:00:03.392Z [INFO]: No custom headers found.
2022-03-29T11:00:03.410Z [WARNING]: !! No index.html detected in deploy folder: /codebuild/output/src244914689/src/scandipwa/packages/scandipwa/build
2022-03-29T11:00:03.410Z [INFO]: # Starting build artifact upload process...
2022-03-29T11:00:03.446Z [INFO]: # Build artifact is: 0MB
2022-03-29T11:00:03.446Z [INFO]: # Uploading build artifact '__artifacts.zip'...
2022-03-29T11:00:03.447Z [INFO]: # Build artifact is: 0MB
2022-03-29T11:00:03.447Z [INFO]: # Uploading build artifact '__artifactsHash.zip'...
2022-03-29T11:00:03.553Z [INFO]: # Build artifact upload completed
2022-03-29T11:00:03.554Z [INFO]: # Starting environment caching...
2022-03-29T11:00:03.554Z [INFO]: # Uploading environment cache artifact...
2022-03-29T11:00:03.612Z [INFO]: # Environment caching completed
Terminating logging...
```

**Comment by zans-laksa** [comment]
It's not a cache issue.
The error log clearly says what is wrong.

image
image

**Comment by florianf22** [comment]
I see, corrected spacing now

File: packages/scandipwa/src/component/CheckoutBilling/CheckoutBilling.style.scss [hunk 1]
```diff
@@ -8,8 +8,8 @@
  * @package scandipwa/base-theme
  * @link https://github.com/scandipwa/base-theme
  */
-
- :root {
+
```
Comment (zans-laksa) [inline packages/scandipwa/src/component/CheckoutBilling/CheckoutBilling.style.scss:11]: Please remove this space too.

**Review by zans-laksa — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/CheckoutBilling/CheckoutBilling.style.scss (hunk 1 above)
Comment (florianf22) [reply packages/scandipwa/src/component/CheckoutBilling/CheckoutBilling.style.scss:11]: space has been removed

File: packages/scandipwa/src/component/CheckoutBilling/CheckoutBilling.component.js [hunk 2]
```diff
@@ -140,6 +143,7 @@ export class CheckoutBilling extends PureComponent {
                   elem="TACLink"
                   onClick={ this.handleShowPopup }
                   type="button"
+                  mods={ { isShifted: screen.width <= SCREEN_WIDTH_THRESHOLD } }
```
[the remark this answers is not in the document]
Comment (florianf22) [reply packages/scandipwa/src/component/CheckoutBilling/CheckoutBilling.component.js:146]: I can't use media queries. there are pre defined media queries for mobile, I didn't find custom media query in the codebase. However, I need to change something when it gets to the very small screen, that's why

**Comment by carinadues** [comment]
@florianf22 There are more changes required. Can you please check?

**Comment by carinadues** [comment]
The issue will be fixed in https://github.com/scandipwa/scandipwa/pull/4709

**Pull request closed without merging**